aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -1148,6 +1157,8 @@ void scatter_impl(
     ReduceStub& reduce_stub,
     FillStub& fill_stub,
     const c10::optional<c10::string_view> reduce = nullopt) {
+  if (index.numel() == 0) return;
+  dim = at::maybe_wrap_dim(dim, self.dim());
   auto mut_out = const_cast<Tensor&>(out);
```

Comment:
Since you propagated `const` for `scatter`, we can get rid of this `const_cast`. It could be done on another PR, though (so as not to block this one).